frontend/catalyst/api_extensions/callbacks.py

```diff
@@ -38,6 +40,64 @@
 from catalyst.utils.types import convert_pytype_to_shaped_array
 
 
+def accelerate(func=None, dev=None):
+    """Seamless integration with jax.jit
+
+    The accelerate function can be used to automatically apply `jax.jit` to
+    the input function `func` and be used to specify the device with the `dev`
+    keyword argument. If `dev` is None, then the default device as determined
+    by JAX is used.
+
+    Can be used in the following modes:
+
+
+    ```py
+    @accelerate
+    def identity(x):
+      return x
+
+    @qml.qjit
+    def qjitted_fn(x):
+      return x
+    ```
+
+    ```
+    @accelerate(dev=jax.devices()[0])
+    def identity(x):
+      return x
+
+    @qml.qjit
+    def qjitted_fn(x):
+      return x
+    ```
+
+    or
+
+    ```
+    @qml.qjit
+    def qjitted_fn(x):
+      return accelerate(lambda x: x)
+    ```
+    """
+
+    if isinstance(func, Callable):
+
+        @functools.wraps(func)
+        def defer(*args, **kwargs):
+            # Make abstract variables from input tracers.
+            absargs, abskwargs = tree_map(shaped_abstractify, (args, kwargs))
+            # Find the shape of the return value
+            _, returnshape = jax.make_jaxpr(func, return_shape=True)(*absargs, **abskwargs)
```

Comment:
if the function has already been jitted, is it possible to get an already generated JAXPR? I guess only if the function has already been traced once?

Reply:
I don't think it is possible to get the jaxpr from a jax.jit-ed function. What I have found is that it is discarded after compilation.

Reply:
Just jumping back in but we don't appear to actually care about the jaxpr itself. Is there no way to get the return type info from the jit decorated object? Would be a shame to trace the function twice for no reason.

Reply:
got it!

Reply:
> Would be a shame to trace the function twice for no reason.

Yep, this is the context behind my question

Reply:
I don't think it is possible.

Reply:
What exactly do you need, would the jaxpr equation provide the required information?

Reply:
If `returnshapes` contains pytrees, can you add a test that verifies they work in the outer scope?

Reply:
See #812 . Since `accelerate` just uses callbacks, it will be the same.